pandas/core/frame.py

```diff
@@ -752,10 +752,13 @@ def _can_fast_transpose(self) -> bool:
         """
         if isinstance(self._mgr, ArrayManager):
             return False
-        if self._mgr.any_extension_types:
-            # TODO(EA2D) special case would be unnecessary with 2D EAs
+        blocks = self._mgr.blocks
+        if len(blocks) != 1:
             return False
-        return len(self._mgr.blocks) == 1
+
+        dtype = blocks[0].dtype
+        # TODO(EA2D): special case would be unnecessary with 2D EAs
+        return isinstance(dtype, np.dtype)  # i.e. not ExtensionDtype
```

Comment:
```suggestion
        return not self._mgr.any_extension_type
```

(at this point you know there is only 1 block, so we are not checking it for potentially many columns as was the case before)

Reply:
This paves the way for the DTBlock/TDBlock-backed-by-DTA/TDA PR, in which this isinstance check is only slightly tweaked

Reply:
I suppose that those blocks should then still not be considered as "extension block" for `any_extension_type`, as the main reason that we use `any_extension_type` is to know if it has a non-consolidating / non-transposable block (and which would not be the case for DTBlock)

Reply:
not worth arguing about, since only one line is in question and that line will get changed in the upcoming PR anyway.  will use the proposed edit.